src/algorand_transactions/transaction.rs

```diff
@@ -491,15 +503,23 @@ impl AlgorandTransaction {
                 Some(app) => app.foreign_assets.clone(),
                 None => None,
             },
-            inner_txs: match &json.inner_txs {
-                Some(inner_txs) => Some(
-                    inner_txs
-                        .iter()
-                        .map(AlgorandTransaction::from_json)
-                        .collect::<Result<Vec<AlgorandTransaction>>>()?,
-                ),
+            inner_txs: match &json.id {
```

Comment:
Maybe I should create a function for this. Is there a way to avoid nested `match`?

Reply:
To be fair the whole `json` handling is full of some iffy `match` statements, and is in need of a good overhaul. Let's keep this as it is for now and we can refactor in a dedicated PR.